river/ensemble/boosting.py

```diff
@@ -118,6 +119,98 @@ def predict_proba_one(self, x):
         return y_proba
 
 
+class ADWINBoostingClassifier(AdaBoostClassifier):
+    """ADWIN Boosting classifier.
+    ADWIN Boosting [^1] is the online boosting method of Oza and Russell [^2]
+    with the addition of the `ADWIN` algorithm as a change detector. If concept
+    drift is detected, the worst member of the ensemble (based on the error
+    estimation by ADWIN) is replaced by a new (empty) classifier.
+    Parameters
+    ----------
+    model
+        The classifier to boost.
+    n_models
+        The number of models in the ensemble.
+    seed
+        Random number generator seed for reproducibility.
+    Examples
+    --------
+    >>> from river import datasets
+    >>> from river import ensemble
+    >>> from river import evaluate
+    >>> from river import linear_model
+    >>> from river import metrics
+    >>> from river import preprocessing
+    >>> dataset = datasets.Phishing()
+    >>> model = ensemble.ADWINBoostingClassifier(
+    ...     model=(
+    ...         preprocessing.StandardScaler() |
+    ...         linear_model.LogisticRegression()
+    ...     ),
+    ...     n_models=3,
+    ...     seed=42
+    ... )
+    >>> metric = metrics.F1()
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    F1: 87.41%
+
+    References
+    ----------
+    [^1]: Albert Bifet, Geoff Holmes, Bernhard Pfahringer, Richard Kirkby,
+    and Ricard Gavaldà. "New ensemble methods for evolving data streams."
+    In 15th ACM SIGKDD International Conference on Knowledge Discovery and
+    Data Mining, 2009.
+    [^2]: Oza, N., Russell, S. "Online bagging and boosting."
+    In: Artificial Intelligence and Statistics 2001, pp. 105–112.
+    Morgan Kaufmann, 2001.
+    """
+
+    def __init__(self, model: base.Classifier, n_models=10, seed: int = None):
+        super().__init__(model, n_models, seed)
+        self._drift_detectors = [copy.deepcopy(drift.ADWIN()) for _ in range(self.n_models)]
```

Comment:
```suggestion
        self._drift_detectors = [drift.ADWIN() for _ in range(self.n_models)]
```

No need to call `copy.clone` if you instantiate the class